src/jaxsim/parsers/descriptions/joint.py

```diff
@@ -14,39 +13,35 @@
 from .link import LinkDescription
 
 
-@enum.unique
-class JointType(enum.IntEnum):
-    """
-    Type of supported joints.
-    """
-
-    @staticmethod
-    def _generate_next_value_(name, start, count, last_values):
-        # Start auto Enum value from 0 instead of 1
-        return count
-
-    #: Fixed joint.
-    F = enum.auto()
+class _JointTypeMeta(type):
+    def __new__(cls, name, bases, dct):
+        cls_instance = super().__new__(cls, name, bases, dct)
 
-    #: Revolute joint (1 DoF around axis).
-    R = enum.auto()
+        # Assign integer values to the descriptors
+        cls_instance.F = 0
+        cls_instance.R = 1
+        cls_instance.P = 2
 
-    #: Prismatic joint (1 DoF along axis).
-    P = enum.auto()
+        return cls_instance
 
 
-@jax_dataclasses.pytree_dataclass
-class JointDescriptor:
+class JointType(metaclass=_JointTypeMeta):
     """
-    Base class for joint types requiring to store additional metadata.
+    Type of supported joints.
     """
 
-    #: The joint type.
-    joint_type: JointType
+    class F:
+        pass
+
+    class R:
+        pass
+
+    class P:
+        pass
```

Comment:
The single letter names are an inheritance from the very first implementation inspired by spatial v2. I'd take advantage of this refactoring to use longer and more explicit names (Fixed, Revolute, Prismatic).

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/153/commits/a9d11e8bf368ee60142ca7e706a3a32220fd4ae6